Network/HTTP/Semantics/FillBuf.hs

```diff
@@ -131,6 +135,7 @@ runStreamingChunk :: StreamingChunk -> NextWithTotal -> NextWithTotal
 runStreamingChunk chunk next =
     case chunk of
         StreamingFinished mdec -> finished mdec
+        StreamingCancelled mErr -> \_total _buf _room -> pure $ CancelNext mErr
```

Comment:
It might be worth adding a comment here indicating why we can ignore `_total`: because we are cancelling this stream, so any partial data sent _on this stream_ so far can be discarded. It's worth being explicit about this since we got it wrong on the `http2` side (where in an earlier version we were discarding all data on the connection, which is of course not correct).

Reply:
Actually, probably useful to add this to the haddocks of `outBodyCancel` also.